### PR TITLE
ONT automated QC reports with ToulligQC

### PR DESCRIPTION
**Pull request by kedhammar**

### CODE REVIEW

**Comment by kedhammar** [comment]
Linked to https://github.com/NationalGenomicsInfrastructure/genomics-status/pull/901

**Comment by kedhammar** [comment]
Necessitates changes to taca.yaml CONFIG["nanopore_analysis"]["toulligqc_reports_dir"], should point to /var/local/toulligqc_reports on ngi-internal.

File: taca/nanopore/ONT_run_classes.py [hunk 1]
```diff
@@ -338,6 +350,113 @@ def copy_html_report(self):
...
+            raise AssertionError(f"No seq data found in {self.run_abspath}")
+        raw_data_format = "pod5" if "pod5" in raw_data_dir else "fast5"
+
+        # Load samplesheet, if any
+        ss_glob = glob.glob(f"{self.run_abspath}/sample_sheet*.csv")
+        if len(ss_glob) == 0:
+            samplesheet = None
+        elif len(ss_glob) > 1:
+            # If multiple samplesheet, use latest one
+            samplesheet = ss_glob.sort()[-1]
```
Comment (alneberg) [inline taca/nanopore/ONT_run_classes.py:383]: Sort would sort them on file name? And the samplesheet file name starts with a timestamp?

File: taca/nanopore/ONT_run_classes.py [hunk 2]
```diff
@@ -338,6 +350,113 @@ def copy_html_report(self):
...
+        # Build command list
+        command_list = [self.toulligqc_executable]
+        for k, v in command_args.items():
+            command_list.append(k)
+            if v:
+                command_list.append(v)
+
+        # Run the command
+        # Small enough to wait for, should be done in 1-5 minutes
+        process = subprocess.run(command_list)
```
Comment (alneberg) [inline taca/nanopore/ONT_run_classes.py:429]: Does it create any files besides the report? Do we need to clean it up sometimes?

File: taca/nanopore/ONT_run_classes.py (hunk 1 above)
Comment (kedhammar) [reply taca/nanopore/ONT_run_classes.py:383]: Yes, they are named w. the following
```
file_name = f"MinKNOW_samplesheet_{process.id}_{TIMESTAMP}_{process.technician.name.replace(' ','')}.csv"
```

File: taca/nanopore/ONT_run_classes.py (hunk 2 above)
Comment (kedhammar) [reply taca/nanopore/ONT_run_classes.py:429]: The tools should create a dir containing the .html report as well as subdirs `data` and `images`. I don't currently see why we'd need cleanup.

Comment (kedhammar) [reply taca/nanopore/ONT_run_classes.py:429]: The tools should create a dir containing the .html report as well as subdirs `data` and `images`. I don't currently see why we'd need cleanup.

**Comment by kedhammar** [comment]
Linked to https://ngi-git.scilifelab.se/NGI/conf/pulls/82

**Comment by kedhammar** [comment]
Linked to https://ngi-git.scilifelab.se/NGI/Documentation/pulls/37

File: taca/nanopore/ONT_run_classes.py (hunk 1 above)
Comment (alneberg) [reply taca/nanopore/ONT_run_classes.py:383]: `process.id` is a LIMS process? That should probably work, even if I'm not entirely comfortable in relying on LIMS internal convention to sort the files.

**Review by alneberg — APPROVED** [review]
Thank you!

**Pull request merged**